FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Models/Jobs/Job.cs

```diff
@@ -167,7 +168,7 @@ public class Job
             FilterInfo filterInfo,
             IEnumerable<PatientWrapper> patients = null,
             int nextTaskIndex = 0,
-            Dictionary<string, TaskContext> runningTasks = null,
+            ConcurrentDictionary<string, TaskContext> runningTasks = null,
             Dictionary<string, int> totalResourceCounts = null,
```

Comment:
Not related to this PR, do we also need to update other dictionaries here? they also be updated by multiple threads..

Reply:
For v0.4.0, I double-check the dictionaries used, only this dictionary is used by multiple threads.

Reply:
Got it, thanks.